### DISCUSSION TITLE
Fetch issue details from github API using personal access token

### DISCUSSION THREAD

**Discussion by parth-kintent** [community/community#30933] [Projects and Issues]
I've been trying to fetch the details for a Github issue using [this](https://docs.github.com/en/rest/issues/issues#get-an-issue) API endpoint using personal access tokens.
Although, it seems I need to give an all inclusive access to the **repo** scope to have this work correctly.

Is there a list of scopes that i can use to have a read-only access for a personal access token to fetch an issue?

Any help is appreciated.

**Comment by jfullstackdev** [comment]
were you given at least a `read` access by the owner of the repo ? if not, then it will not be allowed, I guess so, it's mentioned there, I'm thinking whether you are accessing a private repo of another user

> If the issue was transferred to or deleted from a repository where the authenticated user lacks read access, the API returns a 404 Not Found status. If the issue was deleted from a repository where the authenticated user has read access, the API returns a 410 Gone status. To receive webhook events for transferred and deleted [issues](https://docs.github.com/webhooks/event-payloads/#issues), subscribe to the issues webhook.

**Reply by parth-kintent** [reply]
Hi,

Thanks for the revert. 
About the repo, it's a private repo of which I myself am the owner. And the token is created from the same user.

At the moment, I need to explicitly give access to the complete repo. To gain a read-only access for issues. 

![Github repo access](https://user-images.githubusercontent.com/109530388/186939661-1a28d14d-f0fc-421a-bbf4-092b2bf2da70.png)

**Reply by jfullstackdev** [reply]
in that case, try to check all scopes first and if that works, do the reverse, pin down every scope to see what's working and not, 

anyways you can simply delete every new PAT for specific scopes and for security reasons
